lib/registry/registry-client.ts

```diff
@@ -0,0 +1,620 @@
+/**
```

Comment:
I explored various options and writing our own minimal registry client turned out imo to be the best option.  I wanted to use `node-docker-registry-client` which was the best implementation I found, but that repo appears abandoned and the code is old and stinky.  In the end, I used it somewhat as a guide to write our own.